### PR TITLE
Create image with sbt and Metals initialized

### PR DESCRIPTION
**Pull request by sideeffffect**

### CODE REVIEW

**Comment by sideeffffect** [comment]
This creates a side-car image for Scala as per https://github.com/eclipse/che-plugin-registry/pull/356#issuecomment-606226761

Please have a look that I'm doing the right thing, both from the point of view of Metals (@olafurpg, @tgodzik) and Che (@amisevsk, @ericwill)

The only possible complication here is that the version of the extension is `1.8.6` and I don't have the permissions to create branches in this repo, so I target the branch `1.7.7` which was already pre-created for me.

(sorry it took so long, I was busier than I had thought)

File: .github/workflows/build-publish.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+#
+# Copyright (c) 2019 Red Hat, Inc.
```
Comment (ericwill) [inline .github/workflows/build-publish.yaml]: Year needs to be updated.

File: .github/workflows/pr-check.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+#
+# Copyright (c) 2019 Red Hat, Inc.
```
Comment (ericwill) [inline .github/workflows/pr-check.yaml]: Year needs to be adjusted.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+ENV SBT_VERSION 1.3.10
+
+RUN curl -Ls https://raw.githubusercontent.com/paulp/sbt-extras/master/sbt > /usr/local/bin/sbt && \
+    chmod 0755 /usr/local/bin/sbt && \
+    sbt -sbt-version $SBT_VERSION eval 1+1
+
+ENV METALS_VERSION 0.8.4
+
+RUN curl -Ls https://raw.githubusercontent.com/coursier/coursier/gh-pages/coursier > /usr/local/bin/coursier && \
```
Comment (ericwill) [inline Dockerfile]: These two `RUN` commands can be merged, no?

**Review by ericwill — CHANGES_REQUESTED** [review]
We need an entrypoint into the container, i.e. `entrypoint.sh` and all that comes with it. Take a look at the `Dockerfile` and `etc` directory in: https://github.com/che-dockerfiles/che-sidecar-kubernetes-tooling/tree/1.1.0

**Comment by sideeffffect** [comment]
@ericwill
I've added the entrypoint.sh, hopefully now is everything correct

**Comment by sideeffffect** [comment]
Hi @ericwill , have you had a change to look at this PR since I've amended it? :slightly_smiling_face:

**Comment by ericwill** [comment]
Hi sorry for the delay. I'll take a look later today and get back to you.

File: .github/workflows/build-publish.yaml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+
+name: Build and push container
+
+on:
+  push:
+    branches:
+      - '1.8.6'
```
Comment (ericwill) [inline .github/workflows/build-publish.yaml:15]: This branch has to match the branch in GitHub. It's no big deal -- I'll create a 1.8.6 branch and you can propose a PR against that one.

**Comment by ericwill** [comment]
I've created a new branch [1.8.6](https://github.com/che-dockerfiles/che-sidecar-scala/tree/1.8.6), please propose a new PR against that branch. Other than that it LGTM.

**Pull request closed without merging**